doc/md/compatibility.md

```diff
@@ -1,253 +0,0 @@
-
--   run the initializer for a new stable variable.
-
-Since `Int </: Nat`, the upgrade logic discards the saved `Int` (what if it was `-1`?) and re-runs the initializer instead.
-
-What’s worse, the upgrade silently "succeeded", resetting the counter to `0`.
-
-## Stable type signatures
-
-A stable type signature looks like the "insides" of a Motoko actor type.
-
-For example, `v2`'s stable types:
-
-``` motoko no-repl file=./examples/count-v2.most
-```
-
-An upgrade from `v2` to `v3`'s stable types:
-
-``` motoko no-repl file=./examples/count-v3.most
-```
-
-requires consuming an `Int` as a `Nat`: a ***type error***.
-
-## Dual interface evolution
-
-An upgrade is safe provided:
-
--   the candid interface evolves to a subtype; and
-
--   the stable interface evolves to a compatible one (variable to supertype or new)
-
-Given version `v0` with candid interface `v0.did` and stable type interface `v0.most`:
-
-``` candid file=./examples/count-v0.did
-```
-
-``` motoko no-repl file=./examples/count-v0.most
-```
-
-And version `v1` with candid interface `v1.did` and stable type interface `v1.most`,
-
-``` candid file=./examples/count-v1.did
-```
-
-``` motoko no-repl file=./examples/count-v1.most
-```
-
-And version `v2` with candid interface `v2.did` and stable type interface `v2.most`,
-
-``` candid file=./examples/count-v2.did
-```
-
-``` motoko no-repl file=./examples/count-v2.most
-```
-
-And, finally, version `v3` with candid interface `v3.did` and stable type interface `v3.most`:
-
-``` candid file=./examples/count-v3.did
-```
-
-``` motoko no-repl file=./examples/count-v3.most
-```
-
-The following table summarizes the (in)compatibilities between them:
-
-|         |                  |                       |
-|---------|------------------|-----------------------|
-| version | candid interface | stable type interface |
-| `v0`    | `v0.did`         | `v0.most`             |
-|         | :> ✓             | \<\<: ✓               |
-| `v1`    | `v1.did`         | `v1.most`             |
-|         | :> ✓             | \<\<: ✓               |
-| `v2`    | `v2.did`         | `v2.most`             |
-|         | :> ✓             | \<\<: *✗*             |
-| `v3`    | `v3.did`         | `v3.most`             |
-
-## Tooling
-
-Motoko compiler (`moc`) now supports:
-
--   `moc --stable-types …​` emits stable types to a `.most` file
-
--   `moc --stable-compatible <pre> <post>` checks two `.most` files for upgrade compatibility
-
-To upgrade from `cur.wasm` to `nxt.wasm` we need check both Candid interface and stable variables are "compatible"
-```
-didc check nxt.did cur.did  // nxt <: cur
-moc --stable-compatible cur.most nxt.most  // cur <<: nxt
-```
-E.g. the upgrade from `v2` to `v3` fails this check:
-```
-> moc --stable-compatible v2.most v3.most
-(unknown location): Compatibility error [M0170], stable variable state of previous type
-  var Int
-cannot be consumed at new type
-  var Nat
-```
-
-## Examples in the wild
```

Comment:
The example seems to have been lost in the move but its actually a common case and deserves mention somewhere. Users are constantly trying to add fields to records on upgrade and that's just not sound (because the upgrade would have to insert fields, making up values for them out of thin air).

Reply:
adressed in https://github.com/dfinity/motoko/pull/4512